### ISSUE TITLE
Using hints makes macros to not work

### ISSUE THREAD

**Issue by Hi-Angel** [qutebrowser/qutebrowser#3453]
Texteditor of codinggame.com is one of those where external editor doesn't work, so I wanted to automatize the job of copy-pasting the code and running a game. Unfortunately recorded macro stops on working on the first usage of a hint.

# Steps to reproduce

1. Press on this page <kbd>qq</kbd> *(it starts recoding a macro to `q` register)*.
2. Press <kbd>f</kbd> *(to enumerate links)*.
3. Press keys according to "Pull requests" link.
4. Press <kbd>q</kbd> *(i.e. done writing a macro)*.
5. Return to the previous page, and press <kbd>@q</kbd>

## Expected

Browser repeats your steps by following a link to "Pull requests".

## Actual

Browser only executes the first key of the macro *(i.e. <kbd>f</kbd>)*, then stops.

quitebrowser 1.0.4

**Comment by The-Compiler** [comment]
cc @blyxxyz

**Comment by Hi-Angel** [comment]
FTR: I came up with an idea of a workaround: start recording a macro, enter insert mode with a mouse, type the needed keys, exit insert mode, stop recording a macro. I was sure I gonna find a bug with insert mode not working this way, and triggering everything in normal mode.

Turned out, I was right, but only partially. Qutebrowser still doesn't browse links this way, but I was right that it doesn't try entering insert mode and typing the characters, even though I still have cursor blinking in the textbox 😛

**Comment by The-Compiler** [comment]
See #2668 about that.

**Comment by Hi-Angel** [comment]
No, it's different, my comment is about not working macros with insert mode.

**Comment by eric-foy** [comment]
I needed this and found this works:
`fake-key -g f ;; later 100 fake-key -g _chars_`
where _chars_ is the hint followed

**Comment by lopespt** [comment]
Just voting for this feature 👍

**Comment by The-Compiler** [comment]
@lopespt  Please consider using [the :+1: reaction](https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) in the future, to avoid generating noise in comments as well as notifications for people watching. Thanks!
